### PR TITLE
Documents and Editions Documentation

### PR DESCRIPTION
**Pull request by thomasleese**
Work on improving the documentation following _Documents and Editions_ will go in this PR.

### CODE REVIEW

File: doc/link-expansion.md [hunk 1]
```diff
@@ -279,16 +277,16 @@ By default links contain the following fields:
...
-- `description` - A short description of the content item
-- `document_type` - The document of the content item
-- `locale` - The language the content item linked to is in
-- `public_updated_at` - The date/time that a content item was last changed
-- `schema_name` - The [GOV.UK content schema][govuk-content-schema] that a
-  content item conforms to
-- `title` - The title of a content item
+- `content_id` - A UUID to represents the document
+- `description` - A short description of the content
+- `document_type` - A document type - often identical to schema_name
```
Comment (MatMoore) [inline doc/link-expansion.md]: It would be helpful for this to explain the difference between the two, or link to somewhere else that already does.

In my mind document type influences how we present the document to users (publishers, and to a lesser extent end users) whereas schema name describes data. But the terminology is really confusing (especially considering "format" and "template" are also used a lot in different contexts)

**Comment by MatMoore** [comment]
LGTM - I really like this change to the data model, it makes it much easier to understand.

One question: in https://github.com/alphagov/publishing-api/blob/e35f083e7830088d76f5dc91ec6af6f0cab0f340/doc/model/object-model.png
it shows a `LinkSet` belonging to an `Edition` rather than a `Document`. Is this correct?

I would have expected it to belong to the document because a link is only a relationship between two content ids and doesn't change when you create new editions.

File: doc/link-expansion.md (hunk 1 above)
Comment (kevindew) [inline doc/link-expansion.md]: @MatMoore Sure - A link sounds a good idea. We've struggled explaining document_type in the past as it's quite a loose concept. Would be good to get it defined well in the docs.

**Comment by kevindew** [comment]
Thanks for checking this out @MatMoore and that diagram is incorrect indeed - thanks for pointing that out.

File: README.md [hunk 2]
```diff
@@ -8,91 +8,94 @@ existing piece of content.
...
+- **Content Item**: A representation of content that can be sent to a
+  [content store][content-store].
+- **Link Set**: A collection of links to other documents. It is used to
+  capture relationships between pieces of content (e.g. parent/child)
+- **Unpublishing**: An object indicating a previously published edition
+  which has been removed from the live site.  Can be "gone", "withdrawal", or
+  "redirect".
 - **User**: A user of the system, which is used to restrict what content is
-returned from the API as well as prevent certain actions on content
+  returned from the API as well as prevent certain actions on content.
```
Comment (danielroseman) [inline README.md]: I don't think we restrict get_content by user any more.

File: doc/arch/adr-005-documents-and-editions.md [hunk 3]
```diff
@@ -0,0 +1,106 @@
+# Decision Record: Resolve consistency and uniqueness in content
+
+## Context
+
+Publishing API has adopted the open/closed principal where the domain model is
```
Comment (danielroseman) [inline doc/arch/adr-005-documents-and-editions.md]: typo: principle

File: doc/model.md [hunk 4]
```diff
@@ -1,516 +1,247 @@
...
+[/put-content/:content_id][put-content-api] API.
 
-Content items will be published to the content store regardless of their
-`phase`.  If a content item has a `phase` of either *alpha* or *beta*, a visual
-component will be added to the front-end to show this (assuming the front-end
-application for that format supports this).
+Key fields that are set internally by the Publishing API are:
 
-There is more information on what each of the phases mean [here](https://www.gov.uk/service-manual/phases).
+- `state` - where an edition is in it's [publishing workflow](#workflow), can
```
Comment (danielroseman) [inline doc/model.md]: its

File: doc/model.md [hunk 5]
```diff
@@ -1,516 +1,247 @@
...
+  be "draft", "published", "unpublished" or "superseded".
+- `content_store` - indicates whether an edition is on the draft or live
+  content store, or empty to indicate that it is not on a content store.
```
Comment (danielroseman) [inline doc/model.md]: Could probably do with some more detail here. If something has a value of "live", it may be in both content stores.

File: doc/model.md [hunk 6]
```diff
@@ -1,516 +1,247 @@
...
-We based this object model on the needs of our users. In the Publishing Platform
-team, we are in a unique position such that the majority of our users are in
-fact other developers working on their respective publishing applications.
+A published edition can be unpublished, which will create an
+[`unpublishing`](#unpublishing) for the edition. The unpublished edition will
+be represented on the live content store.
 
-In summary, this object model aims to address the following needs:
+If a draft is published while there is already a published or unpublished
+edition. The previous edition will have it's `state` updated to "superseded"
```
Comment (danielroseman) [inline doc/model.md]: its

File: doc/publishing-application-examples.md [hunk 7]
```diff
@@ -1,30 +1,46 @@
...
 ```
   access_limited : ["8f3173bb-ee2a-48a2-aa86-f2f030830888"]
 ```
 
 ## Draft and Publish endpoint flow
 
-The Publishing API prescribes a draft-to-live workflow where the publishing application submits new draft content, this content may then be updated via the same endpoint and finally the content item can be published in a discreet endpoint request.
+The Publishing API prescribes a draft-to-live workflow where the publishing
+application submits new draft content, this content may then be updated via the
+same endpoint and finally can be published in a discreet
```
Comment (danielroseman) [inline doc/publishing-application-examples.md]: discrete

**Comment by kevindew** [comment]
Thanks @danielroseman - I've addressed all your comments - and made a couple more tweaks.

File: doc/model.md [hunk 8]
```diff
@@ -1,516 +1,253 @@
...
+
+- [Introduction](#introduction)
+  - [content_id](#content_id)
+  - [Diagram](#diagram)
+- [Content](#content)
+  - [Document](#document)
+  - [Edition](#edition)
+    - [Workflow](#workflow)
+    - [Uniqueness](#uniqueness)
+    - [Substitutiion](#substitution)
```
Comment (thomasleese) [inline doc/model.md]: Should be 'Substitution'.

Comment (kevindew) [inline doc/model.md]: Thanks, got it
